### DISCUSSION TITLE
Implementing `basedpyright`'s `reportAny` rule

### DISCUSSION THREAD

**Discussion by etripier** [microsoft/pyright#7940] [Ideas]
Hi Eric, you are the GOAT. Thanks for keeping our Python code type-safe.

Would you accept a PR to implement [`basedpyright`'s `reportAny` rule](https://github.com/DetachHead/basedpyright?tab=readme-ov-file#reportany---fully-ban-the-any-type)? It is incredibly useful for forcing developers to write more specific types instead of taking the easy way out. I'm totally down to open a PR because 1) I'd just be reimplementing an existing solution and that's not too much sweat on my brow and 2) You seem super busy all the time and with summer coming up I'd love to free up your time for a vacation or too. You deserve it!

Anyways, let me know. Devoted Trautgod worshipper here.

**Comment by erictraut** [comment]
Thanks but no thanks. I don't think that `reportAny` is the right approach. There's a good reason that `Any` exists in the type system, and there are many places where it makes sense. If you want to enforce that it doesn't appear in your own code base, that's more of a code style decision that can and should be enforced by a linter. I don't think it's something that should be enforced by a type checker. 

Even if I though this was a good idea, `reportAny` is not a check that could be enabled by default in `strict` mode, and checks that cannot be enabled in `strict` mode are difficult to discover and end up going unused by all but a few pyright users. For that reason, I try to avoid adding any such checks.

**Reply by max-muoto** [reply]
If you wanted to ban imports of `Any` in your codebase, Ruff's `banned-api` configuration would be a good option:  https://docs.astral.sh/ruff/settings/#lint_flake8-tidy-imports_banned-api

**Reply by JonathanPlasse** [reply]
You could use this instead
https://docs.astral.sh/ruff/rules/any-type/
